include/odp/api/spec/traffic_mngr.h

```diff
@@ -278,6 +278,54 @@ typedef struct {
 	 * of the weights allowed at this level. */
 	uint32_t max_weight;
 
+	/** Minimum allowed value for odp_tm_shaper_params_t::commit_burst or
+	 * odp_tm_shaper_params_t::peak_burst when
+	 * odp_tm_shaper_params_t::packet_mode is true.
+	 */
+	uint32_t min_burst_packets;
+
+	/** Maximum allowed value for odp_tm_shaper_params_t::commit_burst or
+	 * odp_tm_shaper_params_t::peak_burst when
+	 * odp_tm_shaper_params_t::packet_mode is true.
+	 */
+	uint32_t max_burst_packets;
+
+	/** Minimum allowed value for odp_tm_shaper_params_t::commit_rate or
+	 * odp_tm_shaper_params_t::peak_rate when
+	 * odp_tm_shaper_params_t::packet_mode is true.
+	 */
+	uint64_t min_rate_packets;
+
+	/** Maximum allowed value for odp_tm_shaper_params_t::commit_rate or
+	 * odp_tm_shaper_params_t::peak_rate when
+	 * odp_tm_shaper_params_t::packet_mode is true.
+	 */
+	uint64_t max_rate_packets;
+
+	/** Minimum allowed value for odp_tm_shaper_params_t::commit_burst or
+	 * odp_tm_shaper_params_t::peak_burst when
+	 * odp_tm_shaper_params_t::packet_mode is false.
+	 */
+	uint32_t min_burst;
+
+	/** Maximum allowed value for odp_tm_shaper_params_t::commit_burst or
+	 * odp_tm_shaper_params_t::peak_burst when
+	 * odp_tm_shaper_params_t::packet_mode is false.
+	 */
+	uint32_t max_burst;
+
+	/** Minimum allowed value for odp_tm_shaper_params_t::commit_rate or
+	 * odp_tm_shaper_params_t::peak_rate when
+	 * odp_tm_shaper_params_t::packet_mode is false.
+	 */
+	uint64_t min_rate;
+
+	/** Maximum allowed value for odp_tm_shaper_params_t::commit_rate or
+	 * odp_tm_shaper_params_t::peak_rate when
+	 * odp_tm_shaper_params_t::packet_mode is false.
+	 */
+	uint64_t max_rate;
```

Comment:
min_rate and max_rate duplicate ODP_TM_MIN_SHAPER_BW and ODP_TM_MAX_SHAPER_BW, raising the question which ones should be used by an application. I suppose the old defines could be deprecated or removed altogether.